pkg/csi_driver/node.go

```diff
@@ -220,6 +227,11 @@ func (s *nodeServer) NodeUnpublishVolume(_ context.Context, req *csi.NodeUnpubli
 	}
 	defer s.volumeLocks.Release(targetPath)
 
+	// Unregister metrics collecter.
+	if s.driver.config.MetricsManager != nil {
+		s.driver.config.MetricsManager.UnregisterMetricsCollector(targetPath)
```

Comment:
is this idempotent as well? (example scenario: a node unpublish retry due to failures)

Reply:
Yes, this call is idempotent.